File: examples/02_dataset.js
/**
 * Crawlingo: Structured data extraction with Dataset.
 *
 * Usage:
 *   npm install crawlingo
 *   node 02_dataset.js
 */

const { Dataset } = require('crawlingo');

async function main() {
  console.log('=== Crawlingo Node.js Dataset Example ===\n');

  const url = 'https://www.rust-lang.org/';
  console.log(`Building dataset query for ${url}...`);

  try {
    const result = await new Dataset(url)
      .autoMatch(true)
      .field('title', 'h1')
      .field('tagline', 'header p')
      .build();

    console.log('\nExtracted Fields:');
    const dict = result.toDict();
    for (const [field, value] of Object.entries(dict)) {
      console.log(`  ${field}: ${value.trim().slice(0, 80)}...`);
    }

    await result.toJson('dataset_result.json');
    await result.toCsv('dataset_result.csv');
    console.log('\nCreated dataset_result.json and dataset_result.csv');
  } catch (error) {
    console.error('Error:', error);
  }
}

main();

File: examples/02_dataset.py
"""Crawlingo: Structured data extraction with Dataset.

Usage:
    pip install crawlingo
    python 02_dataset.py
"""

from crawlingo import Dataset


def main():
    print("=== Crawlingo Structured Dataset Example ===")

    url = "https://httpbin.org/html"
    print(f"Building dataset query for {url}...")

    dataset = (
        Dataset(url)
        .auto_match(True)
        .field("title", "h1")
        .field("author", "p", selector_type="xpath")
        .field("content", "div")
        .build()
    )

    print("\nExtracted Fields:")
    result_dict = dataset.to_dict()
    for field, value in result_dict.items():
        print(f"  {field}: {value[:60]}...")

    print("\nExporting results...")
    dataset.to_json("dataset_result.json")
    dataset.to_csv("dataset_result.csv")
    print("Created 'dataset_result.json' and 'dataset_result.csv'.")


if __name__ == "__main__":
    main()
